systems/plants/RigidBodyManipulator.cpp

```diff
@@ -888,11 +909,32 @@ void RigidBodyManipulator::doKinematics(double* q, bool b_compute_second_derivat
     //throw std::out_of_range(msg + oor.what());
   //}
   //END_DEBUG
+  }
 }
 
 void RigidBodyManipulator::doKinematicsNew(double* q, bool compute_gradients, double* v, bool compute_JdotV) {
+  if (kinematicsInit) {
+    bool skip = true;
+    if (compute_gradients && !gradients_cached)
+      skip = false;
+    if (v && !velocity_kinematics_cached)
+      skip = false;
+    if (compute_JdotV && !jdotV_cached)
+      skip = false;
+    for (int i = 0; i < num_dof; i++) {
+      if (q[i] != cached_q[i] || q[i] != cached_q[i]) {
```

Comment:
Good catch. I actually copied this over from the old doKinematics, so it was wrong there too.